src/borg/testsuite/compress.py

```diff
@@ -157,60 +115,60 @@ def test_auto():
     assert meta["csize"] == len(compressed)
 
 
-def test_obfuscate():
-    compressor = CompressionSpec("obfuscate,1,none").compressor
-    data = bytes(10000)
-    _, compressed = compressor.compress({}, data)
-    assert len(data) <= len(compressed) <= len(data) * 101
-    # compressing 100 times the same data should give at least 50 different result sizes
-    assert len({len(compressor.compress({}, data)[1]) for i in range(100)}) > 50
-
-    cs = CompressionSpec("obfuscate,2,lz4")
-    assert isinstance(cs.inner.compressor, LZ4)
+@pytest.mark.parametrize(
+    "specs, c_type, result_range, obfuscation_factor",
+    [
+        ("obfuscate,1,none", CNONE, 50, 10**1),
+        ("obfuscate,2,lz4", LZ4, 10, 10**2),
+        ("obfuscate,6,zstd,3", ZSTD, 90, 10**6),
+        ("obfuscate,2,auto,zstd,10", Auto, 10, 10**2),
+    ],
+)
+def test_factor_obfuscation(specs, c_type, result_range, obfuscation_factor: int):
```

Comment:
Included a type annotation for obfuscation_factor, as it was showing a warning that these were evaluating to tuples without it.